### PR TITLE
fix(server): add CSP, cors

### PR DESCRIPTION
**Pull request by bodinsamuel**
## Describe your changes

Fixes https://linear.app/nango/issue/NAN-1339/[pen-test]-fix-security-misconfiguration-exposed-in-pen-test
Fixes https://linear.app/nango/issue/NAN-1340/[pen-test]-content-security-policy-csp-header-not-implemented
Fixes https://linear.app/nango/issue/NAN-1454/remediate-medium-severity-vulnerabilities-from-pen-test

- Add helmet.js to fix all notice related to SOC2 
  - CSP
  - No Sniff
  - No embed in iframe
  - HSTS
  
- Setup stricter cors
Allows all origin for public api and expose headers, restrict private api

### CODE REVIEW

File: packages/server/lib/middleware/security.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    app.use(helmet.ieNoOpen());
+    app.use(helmet.frameguard({ action: 'sameorigin' }));
+    app.use(helmet.dnsPrefetchControl());
+    app.use(
+        helmet.hsts({
+            maxAge: 5184000
+        })
+    );
+
+    app.use(
```
Comment (khaliqgant) [inline packages/server/lib/middleware/security.ts]: This intentional?

Comment (bodinsamuel) [reply packages/server/lib/middleware/security.ts]: what is?

Comment (khaliqgant) [reply packages/server/lib/middleware/security.ts]: nvm, misread due to whitespace, ignore!

**Review by khaliqgant — APPROVED** [review]
Looks good. We'll have to remember anytime we add a public facing dependency to add it to the security.ts file. Is there any way we can get notified or automate tests so we know when we're missing something?

**Comment by bodinsamuel** [comment]
> Looks good. We'll have to remember anytime we add a public facing dependency to add it to the security.ts file. Is there any way we can get notified or automate tests so we know when we're missing something?

I just put it there because the code would double the length of the main file, but I don't have strong feeling about this tbh

File: packages/server/lib/middleware/security.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        // == "Content-Security-Policy"
+        helmet.contentSecurityPolicy({
+            reportOnly: reportOnly === 'true' || typeof reportOnly === 'undefined',
```
Comment (TBonnin) [inline packages/server/lib/middleware/security.ts]: is `!(reportOnly === 'false')` not clearer?

File: packages/server/lib/middleware/security.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import { basePublicUrl, baseUrl } from '@nangohq/utils';
+import type { Router } from 'express';
+import helmet from 'helmet';
+
+export function securityMiddleware(app: Router): void {
```
Comment (TBonnin) [inline packages/server/lib/middleware/security.ts]: can we have this function return an array of middlewares than can then be used in the router. ex:
```
app.use(securityMiddlewares)
```

**Review by TBonnin — APPROVED** [review]

File: packages/server/lib/middleware/security.ts (hunk 2 above)
Comment (bodinsamuel) [reply packages/server/lib/middleware/security.ts]: or `reportOnly !== 'false'` :D

File: packages/server/lib/middleware/security.ts (hunk 3 above)
Comment (bodinsamuel) [reply packages/server/lib/middleware/security.ts]: indeed

File: packages/server/lib/middleware/security.ts (hunk 2 above)
Comment (TBonnin) [reply packages/server/lib/middleware/security.ts]: :face_palm:

**Pull request merged**